hoomd/ParticleGroup.h

```diff
@@ -144,9 +144,6 @@ class PYBIND11_EXPORT ParticleGroup
     //! \name Initialization methods
     // @{
 
-    //! Constructs an empty particle group
-    ParticleGroup() : m_num_local_members(0) {};
```

Comment:
An earlier version of this branch constructed an empty particle group through this constructor. Upon debugging, I found that these default constructed particle groups are very broken (cause seg faults). Rather than fixing this constructor, I removed it to find all the places that used it and to prevent such bugs from occurring again. 

An empty ParticleGroup should be initialized with the Null particle selector.